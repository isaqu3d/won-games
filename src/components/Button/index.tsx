const Button = () => (
  <button>
    <h1>Button</h1>
  </button>
)
export default Button
