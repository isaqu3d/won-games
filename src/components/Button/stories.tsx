import { Meta, Story } from '@storybook/react'
import Button from '.'

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    children: {
      control: 'text'
    },
    size: {
      control: { type: 'radio' },
      options: ['small', 'medium', 'large']
    }
  }
} as Meta

const Template: Story = (args) => <Button {...args} />

export const Default = Template.bind({})
Default.args = {
  children: 'Buy now',
  size: 'medium'
}
