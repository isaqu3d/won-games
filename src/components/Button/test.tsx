import { render, screen } from '@testing-library/react'
import Button from '.'

describe('<Button />', () => {
  it('should render the medium size by default', () => {
    render(<Button>Buy Now</Button>)
    expect(screen.getByRole('button', { name: /buy now/i })).toHaveClass(
      'text-base h-10 px-6'
    )
  })

  it('should render the small size', () => {
    render(<Button size="sm">Buy Now</Button>)
    expect(screen.getByRole('button', { name: /buy now/i })).toHaveClass(
      'text-sm h-8 px-4'
    )
  })

  it('should render the large size', () => {
    render(<Button size="lg">Buy Now</Button>)
    expect(screen.getByRole('button', { name: /buy now/i })).toHaveClass(
      'text-lg h-12 px-8'
    )
  })

  it('should render the primary variant', () => {
    render(<Button variant="primary">Buy Now</Button>)
    expect(screen.getByRole('button', { name: /buy now/i })).toHaveClass(
      'bg-gradient-to-b from-red-500 to-primary text-white shadow-md hover:brightness-110 active:brightness-90'
    )
  })
})
