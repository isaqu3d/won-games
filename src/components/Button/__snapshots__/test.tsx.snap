// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the heading 1`] = `
<button>
  <h1>
    Button
  </h1>
</button>
`;
